### PR TITLE
:sparkles: erv2 debug-shell command

### PR DESCRIPTION
**Pull request by chassing**
Add a `qontract-cli external-resources debug-shell` command to create and enter an ERv2 debug environment easily, e.g., this could be used to run `terraform` commands manually.

Example usage:

```
$ qontract-cli --config config.prod.toml external-resources --provision-provider aws --provisioner app-sre-stage --provider rds --identifier glitchtip-dev debug-shell
  Preparing environment ... 0:00:00
bash-5.1# cdktf synth
...
Generated Terraform code for the stacks: CDKTF

bash-5.1# cd cdktf.out/stacks/CDKTF
bash-5.1# ls -l
total 4
-rw-r--r-- 1 root root 3663 Nov 22 13:01 cdk.tf.json

bash-5.1# terraform init

Initializing the backend...
...
Terraform has been successfully initialized!

bash-5.1# terraform plan
Acquiring state lock. This may take a few moments...
random_password.glitchtip-dev-password: Refreshing state... [id=none]
aws_db_instance.glitchtip-dev: Refreshing state... [id=db-HNEHRU73VQYQWNMAXH7VQ5ZBWA]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
```

Ticket: [APPSRE-11274](https://issues.redhat.com/browse/APPSRE-11274)

### CODE REVIEW

**Review by esron — APPROVED** [review]
Just a question, not a blocker. Don't we have a preference of Podman over Docker?

File: tools/qontract_cli.py [hunk 1]
```diff
@@ -4320,6 +4321,43 @@ def migrate(ctx, dry_run: bool, skip_build: bool) -> None:
...
 
 
+@external_resources.command()
+@binary(["docker"])
+@click.pass_context
+def debug_shell(ctx) -> None:
+    """Enter an ERv2 debug shell to manually migrate resources."""
+    # use a temporary directory in $HOME. The MacOS colima default configuration allows docker mounts from $HOME.
+    tempdir = Path.home() / ".erv2-debug"
+    tempdir.mkdir(exist_ok=False)
```
Comment (hemslo) [inline tools/qontract_cli.py:4331]: if any exception throw in `enter_shell`, this temp dir won't be auto deleted, next run will fail here, should we use `tempfile.TemporaryDirectory` and `with` to let it auto delete?

Comment (chassing) [reply tools/qontract_cli.py]: Yes, for sure 🤦 Changed in 009dbe1

**Comment by chassing** [comment]
> Just a question, not a blocker. Don't we have a preference of Podman over Docker?

TBH, I don't know. It depends on your used OS. I'm using Colima (Docker) on my Mac.

**Review by hemslo — APPROVED** [review]
LGTM! And I have both `docker` and `podman` installed on mac, using `podman-desktop` with docker compatibility enabled.

**Pull request merged**